.github/workflows/e2e-sample-apps-embedding-sdk.yml

```diff
@@ -0,0 +1,312 @@
+    name: Check which files changed
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    outputs:
+      e2e_all: ${{ steps.changes.outputs.e2e_all }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test which files changed
+        uses: dorny/paths-filter@v3.0.0
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-paths.yaml
+
+  # if this is a test on a release branch, we need to check the build requirements
+  get-build-requirements:
+    if: |
+      !cancelled() &&
+      contains(github.base_ref || github.ref, 'release-x')
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    outputs:
+      java_version: ${{ fromJson(steps.dependencies.outputs.result).java_version }}
+      node_version: ${{ fromJson(steps.dependencies.outputs.result).node_version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: release
+      - name: Prepare build scripts
+        run: cd ${{ github.workspace }}/release && yarn && yarn build
+      - name: Get build dependencies
+        uses: actions/github-script@v7
+        id: dependencies
+        with:
+          script: | # js
+            const {
+              getBuildRequirements,
+              getVersionFromReleaseBranch,
+            } = require('${{ github.workspace }}/release/dist/index.cjs');
+            const targetBranchName = '${{ github.base_ref || github.ref }}';
+
+            const version = getVersionFromReleaseBranch(targetBranchName);
+            const requirements = getBuildRequirements(version);
+
+            return {
+              java_version: requirements.java,
+              node_version: requirements.node,
+            };
+
+  get-sample-app-compatibility-data:
+    if: |
+      !cancelled()
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    outputs:
+      exclude_sample_app_names: ${{ fromJson(steps.build-sample-app-compatibility-data.outputs.result).exclude_sample_app_names }}
+      sample_app_branch_name: ${{ fromJson(steps.build-sample-app-compatibility-data.outputs.result).sample_app_branch_name }}
+      should_run_tests: ${{ fromJson(steps.build-sample-app-compatibility-data.outputs.result).should_run_tests }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build sample app compatibility data
+        uses: actions/github-script@v7
+        id: build-sample-app-compatibility-data
+        with:
+          script: | # js
+            const ref = process.env.GITHUB_BASE_REF || process.env.GITHUB_REF || "";
+            const branchName = ref.replace(/^refs\/heads\//, "");
+
+            const isMainBranchRef = branchName === 'master';
+            const isReleaseBranchRef = branchName.includes('release-x');
+
+            if (isMainBranchRef) {
+              return {
+                sample_app_branch_name: 'main',
+                should_run_tests: true,
+              };
+            }
+
+            if (isReleaseBranchRef) {
```

Comment:
we already have some of these functionalities implemented, so I'd rather keep using them.

https://github.com/metabase/metabase/blob/8423d6382a9635026918277203ee14ebb09ff942/.github/workflows/build-for-release.yml#L168

you can take a look at those functions in `.github/workflows/build-for-release.yml` and search each function name how they're being used in different workflows.